src/module.c

```diff
@@ -13489,6 +13512,153 @@ int RM_GetDbIdFromDefragCtx(RedisModuleDefragCtx *ctx) {
+    va_start(ap, fmt);
+    argv = moduleCreateSetArgvFromUserFormat(fmt,&argc,ap);
+    va_end(ap);
+
+    /* We handle the above format error only when the client is setup so that
+     * we can free it normally. */
+    if (argv == NULL) {
+        errno = EBADF;
+        return 0;
+    }
+
+    if (key->value == NULL) moduleCreateEmptyKey(key,REDISMODULE_KEYTYPE_SET);
+
+    int i, count = 0;
+    for (i = 0; i < argc; i++) {
+        count += setTypeAdd(key->value,argv[i]->ptr);
+        freeStringObject(argv[i]);
+    }
+    if (count == 0) errno = ENOENT;
+    return count;
+}
+
+/* Remove the specified element from the set and the key will be
+ * removed if has no any element in after remove elements operation.
+ *
+ * * **key**: The set key to be added elements to.
+ * * **fmt**: A format specifier string for the command's arguments. Each
+ *   of the arguments should be specified by a valid type specification.
+ *
+ *     * `c` -- The argument is a pointer to a plain C string (null-terminated).
+ *     * `l` -- The argument is long long integer.
+ *     * `s` -- The argument is a RedisModuleString.
+ *     * `v` -- The argument(s) is a vector of RedisModuleString.
+ * * **...**: The actual arguments to the Redis command.
+ *
+
+ * Return value:
+ *
+ * The number of elements were really deleted.
+ *
+ * If the return value is zero, `errno` is set as follows:
+ *
+ * - EINVAL if any unknown flags are set or if key is NULL.
+ * - ENOTSUP if the key is associated with a non Hash value.
+ * - EBADF if the key was not opened for writing.
+ * - ENOENT if no elements were counted as described under Return value above.
+ *   This is not actually an error. The return value can be zero if the key
+ *   doesn't consist any element to be removed or the key doesn't exist
+ */
+int RM_SetRem(RedisModuleKey *key, const char *fmt, ...) {
+    robj **argv = NULL;
+    int argc = 0;
+    va_list ap;
+
+    if (!key) {
+        errno = EINVAL;
+        return 0;
+    } else if (!key->value) {
+        errno = ENOENT;
+        return 0;
+    } else if (key->value && key->value->type != OBJ_SET) {
+        errno = ENOTSUP;
+        return 0;
+    } else if (!(key->mode & REDISMODULE_WRITE)) {
+        errno = EBADF;
+        return 0;
+    }
+
+    /* Handle arguments. */
+    va_start(ap, fmt);
+    argv = moduleCreateSetArgvFromUserFormat(fmt,&argc,ap);
+    va_end(ap);
+
+    /* We handle the above format error only when the client is setup so that
+     * we can free it normally. */
+    if (argv == NULL) {
+        errno = EBADF;
+        return 0;
+    }
+
+    int i, count = 0;
+    for (i = 0; i < argc; i++) {
+        count += setTypeRemove(key->value,argv[i]->ptr);
+        freeStringObject(argv[i]);
+    }
+    moduleDelKeyIfEmpty(key);
+
+    if (count == 0) errno = ENOENT;
+    return count;
+}
+
+/* Return 1 as member of the key or 0 as not member of the key. */
+int RM_SetIsMember(RedisModuleKey *key, RedisModuleString *ele) {
```

Comment:
i suppose that if we let people avoid creating a RedisModuleString for adding and removing, we should support that for query was well.

Reply:
You mean to use "fmt" method like RM_SetAdd does, right?

Reply:
OK. I got your point. Will change.